platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java

```diff
@@ -390,6 +397,57 @@ boolean onTap(PointF tapPoint) {
     return markerId != NO_ANNOTATION_ID && isClickHandledForMarker(markerId);
   }
 
+  //region Drag Event Handlers
+
+  private @Nullable Marker getMarkerBelowPoint(PointF point) {
+    MarkerHit markerHit = getMarkerHitFromTouchArea(point);
+    long markerId = new MarkerHitResolver(mapboxMap).execute(markerHit);
+    if (markerId != NO_ANNOTATION_ID) {
+      return (Marker) getAnnotation(markerId);
+    }
+    return null;
+  }
+
+  boolean onMoveBegin(PointF focalPoint) {
+    Marker marker = getMarkerBelowPoint(focalPoint);
+    if (marker != null && marker.isDraggable()) {
+      currentMarkerBeingDragged = marker;
+      if (onMarkerDragListener != null) {
+        onMarkerDragListener.onMarkerDragStart(marker);
+      }
+      return true;
+    }
+    return false;
+  }
+
+  boolean onMove(float distanceX, float distanceY) {
+    if (currentMarkerBeingDragged == null || !currentMarkerBeingDragged.isDraggable()) {
+      return false;
+    }
+    Projection projection = mapboxMap.getProjection();
+    PointF screenLocation = projection.toScreenLocation(currentMarkerBeingDragged.getPosition());
+    screenLocation.offset(-distanceX, -distanceY);
+    currentMarkerBeingDragged.setPosition(projection.fromScreenLocation(screenLocation));
+
+    if (onMarkerDragListener != null) {
+      onMarkerDragListener.onMarkerDrag(currentMarkerBeingDragged);
+    }
+    return true;
+  }
+
+  boolean onMoveEnd() {
```

Comment:
I feel like `onMoveEnd` logic should be moved to a separate method and invoked when there are more than 1 pointers placed on the screen to finish the drag.